buyer-api/src/queues/notarizationQueue.js

```diff
@@ -1,11 +1,96 @@
+import uuid from 'uuid/v4';
 import { createQueue } from './createQueue';
-import { logger } from '../utils';
+import {
+  dataResponses,
+  dataResponsesBatches as batches,
+  notarizations,
+} from '../utils/stores';
+import logger from '../utils/logger';
+import config from '../../config';
+
+const { buyerPublicBaseUrl } = config;
+
+/**
+ * @async
+ * @function createNotarizationRequest
+ *  Creates a Notarization record containing the NotarizationRequest.
+ * @param {string} notaryAddress Notary's Ethereum address
+ * @param {number} orderId Order ID in the DataExchange contract
+ * @param {import('../utils/stores').NotarizationSeller[]} sellers List of sellers data
+ * @returns {string} ID of the Notarization record
+ */
+const createNotarizationRequest = (notaryAddress, orderId, sellers) => {
+  const id = uuid();
+  const callbackUrl = `${buyerPublicBaseUrl}/notarization-result/${id}`;
+  const request = {
+    orderId, sellers, callbackUrl, status: 'created',
+  };
+  notarizations.store(id, { notaryAddress, request });
+  return id;
+};
+
+/**
+ * @async
+ * @function collectNotarizationSellers
+ *  Fetches every DataResponse to extract the fields required for notarization.
+ * @param {string[]} dataResponseIds List of DataResponses IDs
+ * @returns {import('../utils/stores').NotarizationSeller[]} List of sellers data
+ */
+const collectNotarizationSellers = async dataResponseIds =>
+  dataResponseIds.reduce(async (accumulatorPromise, dataResponseId) => {
+    const accumulator = await accumulatorPromise;
+    const {
+      sellerAddress,
+      sellerId,
+      decryptionKeyHash,
+    } = await dataResponses.fetch(dataResponseId);
+    return [...accumulator, { sellerAddress, sellerId, decryptionKeyHash }];
+  }, Promise.resolve([]));
 
 const queue = createQueue('NotarizationQueue');
 
-export const prepare = async ({ id }) => {
-  logger.info(`N[${id}] :: Prepare :: fake implementation`);
-  return true;
+export const addPrepareNotarizationJob = params => queue.add('prepare', params);
+export const addRequestNotarizationJob = params => queue.add('request', params);
+
+/**
+ * @async
+ * @function prepare
+ *  Builds the NotarizationRequest object from the DataResponse Batch and
+ *  enqueues another job to send the request to the notary.
+ * @param {number} job.id
+ * @param {string} job.data.batchId Batch ID from where the Notarization is built.
+ * @returns {string} The Notarization ID
+ */
+export const prepare = async ({ id, data: { batchId } }) => {
+  const {
+    orderId,
+    notaryAddress,
+    dataResponseIds,
+    status,
+    ...rest
+  } = await batches.fetch(batchId);
+
+  if (status !== 'created') {
+    logger.warn(`N[${id}] :: Prepare :: Can't prepare notarization (${status})`);
+    return null;
+  }
+
+  const sellers = await collectNotarizationSellers(dataResponseIds);
```

Comment:
```suggestion
  const sellers =  dataResponseIds => Promise.all(dataResponseIds.map(dataResponses.fetch)
```

Reply:
Having seen this branch https://github.com/wibsonorg/buyer-sdk/tree/ns-level-performance-test I'm more than ok to go with this approach. Thanks @Meketh, great work!